### ISSUE TITLE
Meteorological fields and boundary conditions read every hour in 14.4.0

### ISSUE THREAD

**Issue by nicholasbalasus** [geoschem/geos-chem#2344]
### Your name

Nick Balasus

### Your affiliation

Harvard University

### What happened? What did you expect to happen?

Running the nested CH4 simulation out of the box for 14.4.0, I find that meteorological fields and boundary conditions are read every hour of the simulation when they should only be read at the beginning of each simulation day given the file structures. This did not happen in 14.3.1. For the nested simulation, reading the met fields 23 extra times per day slows the simulation down significantly.

Two hours from `GC.log` for 14.3.1:
```
---> DATE: 2019/01/01  UTC: 01:00
     - Found all A1     met fields for 2019/01/01 01:30
---> DATE: 2019/01/01  UTC: 01:05
---> DATE: 2019/01/01  UTC: 01:10
---> DATE: 2019/01/01  UTC: 01:15
---> DATE: 2019/01/01  UTC: 01:20
---> DATE: 2019/01/01  UTC: 01:25
---> DATE: 2019/01/01  UTC: 01:30
---> DATE: 2019/01/01  UTC: 01:35
---> DATE: 2019/01/01  UTC: 01:40
---> DATE: 2019/01/01  UTC: 01:45
---> DATE: 2019/01/01  UTC: 01:50
---> DATE: 2019/01/01  UTC: 01:55
     - Creating file for StateMet; reference = 20190101 020000
        with filename = OutputDir/GEOSChem.StateMet.20190101_0200z.nc4
     - Creating file for SpeciesConc; reference = 20190101 020000
        with filename = OutputDir/GEOSChem.SpeciesConc.20190101_0200z.nc4
---> DATE: 2019/01/01  UTC: 02:00
     - Found all A1     met fields for 2019/01/01 02:30
---> DATE: 2019/01/01  UTC: 02:05
```

Two hours from `GC.log` for 14.4.0:
```
---> DATE: 2019/01/01  UTC: 01:00
HEMCO: Opening /n/holyscratch01/external_repos/GEOS-CHEM/gcgrid/gcdata/ExtData/GEOS_0.25x0.3125_NA/GEOS_FP/2019/01/GEOSFP.20190101.A1.025x03125.NA.nc
HEMCO: Opening /n/holyscratch01/external_repos/GEOS-CHEM/gcgrid/gcdata/ExtData/GEOS_0.25x0.3125_NA/GEOS_FP/2019/01/GEOSFP.20190101.A3cld.025x03125.NA.nc
HEMCO: Opening /n/holyscratch01/external_repos/GEOS-CHEM/gcgrid/gcdata/ExtData/GEOS_0.25x0.3125_NA/GEOS_FP/2019/01/GEOSFP.20190101.A3dyn.025x03125.NA.nc
HEMCO: Opening /n/holyscratch01/external_repos/GEOS-CHEM/gcgrid/gcdata/ExtData/GEOS_0.25x0.3125_NA/GEOS_FP/2019/01/GEOSFP.20190101.A3mstC.025x03125.NA.nc
HEMCO: Opening /n/holyscratch01/external_repos/GEOS-CHEM/gcgrid/gcdata/ExtData/GEOS_0.25x0.3125_NA/GEOS_FP/2019/01/GEOSFP.20190101.A3mstE.025x03125.NA.nc
HEMCO: Opening /n/holyscratch01/external_repos/GEOS-CHEM/gcgrid/gcdata/ExtData/GEOS_0.25x0.3125_NA/GEOS_FP/2019/01/GEOSFP.20190101.I3.025x03125.NA.nc
HEMCO: Opening /n/holylfs05/LABS/jacob_lab/Everyone/imi/ch4/blended-boundary-conditions/v2024-06/GEOSChem.BoundaryConditions.20190101_0000z.nc4
     - Found all A1     met fields for 2019/01/01 01:30
---> DATE: 2019/01/01  UTC: 01:05
---> DATE: 2019/01/01  UTC: 01:10
---> DATE: 2019/01/01  UTC: 01:15
---> DATE: 2019/01/01  UTC: 01:20
---> DATE: 2019/01/01  UTC: 01:25
---> DATE: 2019/01/01  UTC: 01:30
---> DATE: 2019/01/01  UTC: 01:35
---> DATE: 2019/01/01  UTC: 01:40
---> DATE: 2019/01/01  UTC: 01:45
---> DATE: 2019/01/01  UTC: 01:50
---> DATE: 2019/01/01  UTC: 01:55
     - Creating file for StateMet; reference = 20190101 020000
        with filename = OutputDir/GEOSChem.StateMet.20190101_0200z.nc4
     - Creating file for SpeciesConc; reference = 20190101 020000
        with filename = OutputDir/GEOSChem.SpeciesConc.20190101_0200z.nc4
---> DATE: 2019/01/01  UTC: 02:00
HEMCO: Opening /n/holyscratch01/external_repos/GEOS-CHEM/gcgrid/gcdata/ExtData/GEOS_0.25x0.3125_NA/GEOS_FP/2019/01/GEOSFP.20190101.A1.025x03125.NA.nc
HEMCO: Opening /n/holyscratch01/external_repos/GEOS-CHEM/gcgrid/gcdata/ExtData/GEOS_0.25x0.3125_NA/GEOS_FP/2019/01/GEOSFP.20190101.A3cld.025x03125.NA.nc
HEMCO: Opening /n/holyscratch01/external_repos/GEOS-CHEM/gcgrid/gcdata/ExtData/GEOS_0.25x0.3125_NA/GEOS_FP/2019/01/GEOSFP.20190101.A3dyn.025x03125.NA.nc
HEMCO: Opening /n/holyscratch01/external_repos/GEOS-CHEM/gcgrid/gcdata/ExtData/GEOS_0.25x0.3125_NA/GEOS_FP/2019/01/GEOSFP.20190101.A3mstC.025x03125.NA.nc
HEMCO: Opening /n/holyscratch01/external_repos/GEOS-CHEM/gcgrid/gcdata/ExtData/GEOS_0.25x0.3125_NA/GEOS_FP/2019/01/GEOSFP.20190101.A3mstE.025x03125.NA.nc
HEMCO: Opening /n/holyscratch01/external_repos/GEOS-CHEM/gcgrid/gcdata/ExtData/GEOS_0.25x0.3125_NA/GEOS_FP/2019/01/GEOSFP.20190101.I3.025x03125.NA.nc
HEMCO: Opening /n/holylfs05/LABS/jacob_lab/Everyone/imi/ch4/blended-boundary-conditions/v2024-06/GEOSChem.BoundaryConditions.20190101_0000z.nc4
     - Found all A1     met fields for 2019/01/01 02:30
---> DATE: 2019/01/01  UTC: 02:05
```

It isn't shown here, but I had a similar experience for a global 2 x 2.5 CH4 simulation.

### What are the steps to reproduce the bug?

```
#!/bin/bash

source /n/home06/nbalasus/envs/gcclassic.rocky+gnu10.minimal.env
dir="${SCRATCH}/jacob_lab/${USER}/testing-gc"
rm -rf "${dir}"

for version in 14_4_0 14_3_1; do

    # Make a directory for this version of GC
    run_dir="${dir}/${version}"
    mkdir -p "${run_dir}"
    cd "${run_dir}"

    # Compile GCClassic for CH4/GEOS-FP/0.25x0.3125/NA/47 L
    git clone https://github.com/geoschem/GCClassic.git
    cd GCClassic
    git checkout ${version//_/.}
    git submodule update --init --recursive
    cd run
    gc_dir="gc"
    if [ $version = 14_4_0 ]; then
        c="9\n2\n4\n4\n2\n${run_dir}\n${gc_dir}\nn\n"
    elif [ $version = 14_3_1 ]; then
        c="3\n2\n4\n4\n2\n${run_dir}\n${gc_dir}\nn\n"
    fi
    printf ${c} | ./createRunDir.sh
    cd "${run_dir}/${gc_dir}/build"
    cmake ../CodeDir -DRUNDIR=..
    make -j
    make install
    cd "${run_dir}/${gc_dir}"

    # Run for 1 day, read the IMI boundary conditions, and use cropped met fields
    sed -i -e "s|20190201|20190102|g" geoschem_config.yml
    sed -i -e "s|\$ROOT/SAMPLE_BCs/v2021-07/CH4/|/n/holylfs05/LABS/jacob_lab/Everyone/imi/ch4/blended-boundary-conditions/v2024-06/|g" HEMCO_Config.rc
    sed -i -e "s|00000100 000000|00000000 010000|g" \
        -e "s|'CH4',|#'CH4',|g" \
        -e "s|'Metrics',|#'Metrics',|g" \
        -e "s|time-averaged|instantaneous|g" HISTORY.rc
    sed -i -e "s|GEOS_0.25x0.3125|GEOS_0.25x0.3125_NA|g" \
        -e "s|\$RES|\$RES.NA|g" HEMCO_Config.rc.gmao_metfields

    # Modify the run script then submit it
    cp runScriptSamples/operational_examples/harvard_cannon/geoschem.run .
    sed -i -e "s|-c 8|-c 32|g" \
        -e "s|15000|64000|g" \
        -e "s|sapphire,huce_cascade,seas_compute,shared|huce_ice|g" geoschem.run
    sbatch geoschem.run

done
```

### Please attach any relevant configuration and log files.

[GC.log.14.3.1.txt](https://github.com/user-attachments/files/15932333/GC.log.14.3.1.txt)
[GC.log.14.4.0.txt](https://github.com/user-attachments/files/15932332/GC.log.14.4.0.txt)

### What GEOS-Chem version were you using?

14.4.0

### What environment were you running GEOS-Chem on?

Local cluster

### What compiler and version were you using?

gcc/10.2.0

### Will you be addressing this bug yourself?

Yes, but I will need some help

### In what configuration were you running GEOS-Chem?

GCClassic

### What simulation were you running?

CH4

### As what resolution were you running GEOS-Chem?

0.25x0.3125

### What meterology fields did you use?

GEOS-FP

**Comment by nicholasbalasus** [comment]
I believe this is coming from #2263.

To avoid the maintenance problem described in #2263 but fix the problem here, could we instead just use a large date range like `1980-2050` for these boundary condition and met field entries?

I tested this above by adding:
```
    if [ $version = 14_4_0 ]; then
        sed -i -e "s|SpeciesBC_?ADV?  \$YYYY/\$MM/\$DD/|SpeciesBC_?ADV?  1980-2050/1-12/1-31/|g" HEMCO_Config.rc
        sed -i -e "s|\$YYYY/\$MM/\$DD/|1980-2050/1-12/1-31/|g" \
            -e "s|\$YYYY/\$MM/1-31|1980-2050/1-12/1-31|g" HEMCO_Config.rc.gmao_metfields
    fi
```
and it worked.

What do you think @msulprizio @yantosca @lizziel @jimmielin?

**Comment by yantosca** [comment]
Thanks @nicholasbalasus.  I think the large date range would work.  I would suggest using a range of 1900-2100 to accommodate GCAP simulations.  Also maybe put a comment in the `HEMCO_Config.rc` file to denote this.

**Comment by nicholasbalasus** [comment]
Thank you @yantosca! I can make a PR for this. 

This should speed up all simulations, both global and nested. In my opinion, the slowdown from this issue makes using 14.4.0 not feasible for a simulation of any substantial length (at least in my use cases). Given this, do you think there is the possibility for a quick 14.4.1 patch?

**Comment by yantosca** [comment]
Hi @nicholasbalasus, yes we can make a 14.4.1 patch version.  We have a couple other updates that can go in soon.

**Comment by msulprizio** [comment]
Thank you @nicholasbalasus for catching and fixing this!

**Comment by yantosca** [comment]
We have merged PR #2347 into the GEOS-Chem "no-diff-to-benchmark" development stream.
